libs/core/src/lib/splitter/splitter-pane-container/splitter-pane-container.component.ts

```diff
@@ -0,0 +1,439 @@
+        this._cdr.markForCheck();
+    }
+
+    /** @hidden */
+    _onPageChange(page: string = ROOT_PAGE): void {
+        this._currentPage = page;
+
+        this._cdr.markForCheck();
+    }
+
+    /** @hidden */
+    private _updatePages(): void {
+        this._pages = [];
+
+        const offCanvasDemandPages = this._panes.filter(pane => !pane.isOnCanvas && pane.demandPane).map(pane => pane.id);
+        let newPage = ROOT_PAGE;
+
+        if (offCanvasDemandPages.length) {
+            this._pages = offCanvasDemandPages;
+            newPage = offCanvasDemandPages[0];
+
+            if (this._panesOnCanvas.length > 1) {
+                this._pages.unshift(ROOT_PAGE);
+                newPage = ROOT_PAGE;
+            }
+
+            if (this._panesOnCanvas.length === 1) {
+                this._pages.unshift(this._panesOnCanvas[0].id);
+                newPage = this._panesOnCanvas[0].id;
+            }
+        }
+
+        const currentPageExist = this._pages.includes(this._currentPage);
+
+        if (!currentPageExist) {
+            /** Detaching basically works without the code but it's the edge-case */
+            const isShouldDetachDefaultPane = this._currentPage === this._defaultPane?.id && newPage === ROOT_PAGE;
+
+            if (isShouldDetachDefaultPane && this._defaultPane?._content.isAttached) {
+                this._defaultPane._content.detach();
+            }
+
+            this._onPageChange(newPage);
+            return;
+        }
+
+        this._cdr.markForCheck();
+    }
+
+    /** @hidden */
+    private _setPanes(): void {
+        this._resizePanesToFitInContainer(true);
+
+        const setDirectPanesSubscription = () => {
+            this._directPaneSubscription$.unsubscribe();
+            this._directPaneSubscription$ = new Subscription();
+
+            this._directPanes.forEach(pane => {
+                this._directPaneSubscription$.add(
+                    pane.toggleOnCanvas.subscribe(() => this._resizePanesToFitInContainer())
+                );
+            });
+        }
+
+        let prevDirectPanes = [...this._directPanes];
+        const areDirectPanesSame = (panes: QueryList<SplitterSplitPaneComponent>): boolean => {
+            if (panes.length !== prevDirectPanes.length) {
+                return false;
+            }
+
+            return prevDirectPanes.every(prevPane => !!panes.find(pane => pane.id === prevPane.id));
+        }
+
+        setDirectPanesSubscription();
+
+        this._subscription$.add(
+            this._panes.changes
+                .pipe(filter(panes => !areDirectPanesSame(panes)))
+                .subscribe(panesList => {
+                    prevDirectPanes = panesList.toArray();
+
+                    setDirectPanesSubscription()
+
+                    this._resizePanesToFitInContainer(true);
+                })
+        )
+
+        if (this._document) {
+            this._subscription$.add(
+                fromEvent(this._document.defaultView, 'resize')
```

Comment:
If you set listener to the document, angular will run change detection on resize. You should to set listener outOfAngular zone or use `ViewportRuler`.